### ISSUE TITLE
Calling unknown method (Campaign integration)

### ISSUE THREAD

**Issue by redkrupp** [verbb/formie#181]
![s6LdxEB](https://user-images.githubusercontent.com/1551388/96336393-45195800-10a1-11eb-9295-504cb03eb18c.png)

Job Data

{
    "submissionId": 43889,
    "integration": {
        "listId": "43879",
        "optInField": "{enterYourEmailToSeeYourResults}",
        "fieldMapping": {
            "email": "{enterYourEmailToSeeYourResults}"
        },
        "name": "Campaign",
        "handle": "campaign",
        "type": null,
        "enabled": "1",
        "sortOrder": "1",
        "cache": {
            "settings": {
                "lists": [
                    {
                        "class": "verbb\\formie\\models\\IntegrationCollection",
                        "id": "43879",
                        "name": "Symptom Checker Results",
                        "fields": [
                            {
                                "class": "verbb\\formie\\models\\IntegrationField",
                                "handle": "email",
                                "name": "Email",
                                "type": null,
                                "required": true,
                                "options": []
                            }
                        ]
                    }
                ]
            }
        },
        "tokenId": null,
        "uid": "af8ee0b8-f417-4932-8280-d090305ee0ad",
        "id": "1",
        "dateCreated": "2020-10-17T01:29:18-08:00",
        "dateUpdated": "2020-10-17T03:42:07-08:00"
    },
    "description": null
}

Campaign 1.16.3 Pro
Formie 1.2.23.1

**Comment by engram-design** [comment]
Just fixed for the next release.

To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.2.23.1",
  "...": "..."
}
```

Then run `composer update`.

**Comment by redkrupp** [comment]
Updated to the dev branch.

![76EN5Uc](https://user-images.githubusercontent.com/1551388/96374546-30b68780-1195-11eb-8e8d-feec9221ce91.png)

**Comment by engram-design** [comment]
I can't seem to replicate that error. The only place we use `userAgent` information is when populating a hidden field - if you've chosen to use the userAgent as the hidden field value. But this is also protected by checks against whether it's a console request.

Are you doing anything with hidden fields?

**Comment by redkrupp** [comment]
Nope. I have double checked the form and campaign setting. No hidden fields there.

**Comment by engram-design** [comment]
Did you re-run the same failed task, or is this a new task? Just trying to figure out where `getUserAgent` is getting called from.

**Comment by redkrupp** [comment]
This happens every time the form is getting submitted (and email address provided). Re-running the task won't do anything. I've also installed freeform pro, duplicated the form and tested it with campaign. It works with no issues.

**Comment by redkrupp** [comment]
I'll use the preinstalled contact form and make an integration with campaign there with no new fields added. I'll let you know in a moment.

**Comment by engram-design** [comment]
Quick one, I assume you're triggering your queue from the command line? I think it might actually be a Campaign plugin issue - https://github.com/putyourlightson/craft-campaign/blob/b45de549f53095e36f2b3c9e7ccf53790f807f60/src/helpers/ContactActivityHelper.php#L71

The reason its not an issue in Freeform is they don't use queue jobs to trigger integrations.

**Comment by engram-design** [comment]
@EmptyStreets Can confirm if processing your queue via the console this will throw an issue. See https://github.com/putyourlightson/craft-campaign/pull/187

**Comment by redkrupp** [comment]
I don't use command line for this. I've just created a form with email field and trying to subscribe the users using campaign integration. The only non standard thing here is that I'm using redis for queue management and async queue plugin. I'll revert everything to defaults and try agian.

**Comment by engram-design** [comment]
It's possible the async queue plugin is spawning processes that could be considered console commands - I'm not entirely sure as I haven't used that plugin. Wondering if it's easy enough for you to test that PR to Campaign to see if that helps?

**Comment by redkrupp** [comment]
I can confirm. Async queue was the reason for everything. redis for queue management works well. Not sure about the console issue above though.

**Comment by engram-design** [comment]
Should be fixed in [1.2.24](https://github.com/verbb/formie/releases/tag/1.2.24), and https://github.com/putyourlightson/craft-campaign/pull/187
